### PR TITLE
Open window to set a username when empty

### PR DESCRIPTION
**Pull request by KeziahMoselle**

### CODE REVIEW

**Comment by KeziahMoselle** [comment]
Change icon from `icon.load` to `icon.fail` when no username : 
## Before

## After


I think it's better to show this icon because it warns the user that he has to perform an action.

**Comment by KeziahMoselle** [comment]
Tested on MacOS too

**Comment by jamieweavis** [comment]
Good idea, much better user experience after installation.

File: src/app.js [hunk 1]
```diff
@@ -127,6 +126,12 @@ app.on('ready', () => {
 
     const username = store.get('username');
 
+    if (username === '') {
```
Comment (jamieweavis) [inline src/app.js]: Could we move this check up a few lines before we set the application state to Loading?

We will also need to create a context menu, otherwise it will never be created. I think the best solution would be to create it with `0`'s as the values for now - in the future it would be nice to do an object-oriented refactor with proper states.

**Review by jamieweavis — CHANGES_REQUESTED** [review]
Good idea, much better user experience after installation.

**Review by jamieweavis — APPROVED** [review]

**Review by jamieweavis — CHANGES_REQUESTED** [review]
I've just realised that the `setInterval` at the bottom will need to be moved now - otherwise the username window will pop up every 15 minutes if unset.

It should probably be moved to just after the state has been changed to loading, it should also be changed to a `setTimeout` since `requestContributionData` is called every 15 minutes and we don't want to end up with dozens of intervals accumulating.

**Comment by KeziahMoselle** [comment]
Can I move the check outside the `requestConstributionData()` function ? So the check will only trigger when the app is launched ?

**Comment by jamieweavis** [comment]
Yeah, that would also work 👍

**Comment by KeziahMoselle** [comment]
I will try that after my classes :+1:

File: src/app.js [hunk 2]
```diff
@@ -181,11 +171,20 @@ app.on('ready', () => {
...
+    return;
+  }
   
   app.on('window-all-closed', () => {});
   tray.on('right-click', requestContributionData);
   ipcMain.on('setUsername', setUsername);
 
   requestContributionData();
-  setInterval(requestContributionData, 1000 * 60 * 15); // 15 Minutes
+  setInterval(requestContributionData, 10000); // 15 Minutes
```
Comment (jamieweavis) [inline src/app.js]: Why was this changed?

Comment (KeziahMoselle) [reply src/app.js]: I undo it in the next commit, because of debug purpose and forgot to change the value, sorry

Comment (jamieweavis) [reply src/app.js]: Ah, sorry I didn't see that you changed it back!

**Review by jamieweavis — APPROVED** [review]

**Comment by jamieweavis** [comment]
Thanks again for this, I will look at doing the `v1.1.0` release tomorrow 👍

**Comment by KeziahMoselle** [comment]
No problem ! Great :tada:

**Comment by jamieweavis** [comment]
Just to let you know, I've had to change it back so that the check is in `requestContributionData`.

I didn't notice last night, but returning early in the app ready function isn't good practice and was causing issues.

**Comment by KeziahMoselle** [comment]
Oh no problem, sorry I didn't thought of that...
